examples/applications/plot_time_series_lagged_features.py

```diff
@@ -1,425 +1,435 @@
+
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores = consolidate_scores(cv_results, scores, metric)
+
+
+# %%
+# Modeling predictive uncertainty via quantile regression
+# -------------------------------------------------------
+# Instead of modeling the expected value of the distribution of
+# :math:`Y|X` like the least squares and Poisson losses do, one could try to
+# estimate quantiles of the conditional distribution.
+#
+# :math:`Y|X=x_i` is expected to be a random variable for a given data point
+# :math:`x_i` because we expect that the number of rentals cannot be 100%
+# accurately predicted from the features. It can be influenced by other
+# variables not properly captured by the existing lagged features. For
+# instance whether or not it will rain in the next hour cannot be fully
+# anticipated from the past hours bike rental data. This is what we
+# call aleatoric uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+quantile_list = [0.05, 0.5, 0.95]
+
+for quantile in quantile_list:
+    model = HistGradientBoostingRegressor(loss="quantile", quantile=quantile)
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=ts_cv,
+        scoring=scoring,
+        n_jobs=2,
+    )
+    time = cv_results["fit_time"]
+    scores["fit_time"].append(f"{time.mean():.2f} ± {time.std():.2f} s")
+
+    scores["loss"].append(f"quantile {int(quantile*100)}")
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            metric = key.split("test_")[1]
+            scores = consolidate_scores(cv_results, scores, metric)
+
+scores_df = pl.DataFrame(scores)
+scores_df
+
+
+# %%
+# Let us take a look at the losses that minimise each metric.
+def min_arg(col):
+    col_split = pl.col(col).str.split(" ")
+    return pl.arg_sort_by(
+        col_split.list.get(0).cast(pl.Float64),
+        col_split.list.get(2).cast(pl.Float64),
+    ).first()
+
+
+scores_df.select(
+    pl.col("loss").get(min_arg(col_name)).alias(col_name)
+    for col_name in scores_df.columns
+    if col_name != "loss"
+)
+
+# %%
+# Even if the score distributions overlap due to the variance in the dataset,
+# it is true that the average RMSE is lower when `loss="squared_error"`, whereas
+# the average MAPE is lower when `loss="absolute_error"` as expected. That is
+# also the case for the Mean Pinball Loss with the quantiles 5 and 95. The score
+# corresponding to the 50 quantile loss is overlapping with the score obtained
+# by minimizing other loss functions, which is also the case for the MAE.
+#
+# A qualitative look at the predictions
+# -------------------------------------
+# We can now visualize the performance of the model with regards
+# to the 5th percentile, median and the 95th percentile:
```

Comment:
this is where I wish we had much nicer / easier API to get the same output with much less boilerplate.

Reply:
We could use a for loop instead of repeating calls. However this is outside of the scope of this PR.